rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -246,77 +247,77 @@ public String defineNodeSource(String sessionId, String nodeSourceName, String i
+                                                                                                                   nodeSourceName,
+                                                                                                                   infrastructureType,
+                                                                                                                   infrastructureParameters,
+                                                                                                                   infrastructureFileParameters,
+                                                                                                                   policyType,
+                                                                                                                   policyParameters,
+                                                                                                                   policyFileParameters));
     }
 
     @Override
     public String deployNodeSource(String sessionId, String nodeSourceName)
             throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(restClient -> restClient.deployNodeSource(sessionId,
-                                                                                             nodeSourceName));
+        return executeFunctionReturnStreamAsStringWithoutNewLines(restClient -> restClient.deployNodeSource(sessionId,
+                                                                                                            nodeSourceName));
     }
 
     @Override
     public String undeployNodeSource(String sessionId, String nodeSourceName, boolean force)
             throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(restClient -> restClient.undeployNodeSource(sessionId,
-                                                                                               nodeSourceName,
-                                                                                               force));
+        return executeFunctionReturnStreamAsStringWithoutNewLines(restClient -> restClient.undeployNodeSource(sessionId,
+                                                                                                              nodeSourceName,
+                                                                                                              force));
     }
 
     @Override
     public String getInfrastructures(String sessionId) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(restClient -> restClient.infrastructures(sessionId));
+        return executeFunctionReturnStreamAsStringWithoutNewLines(restClient -> restClient.infrastructures(sessionId));
     }
 
     @Override
     public String getPolicies(String sessionId) throws RestServerException, ServiceException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)